tests/system/src/test/java/test/robot/javafx/scene/RobotTest.java

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright (c) 2018, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package test.robot.javafx.scene;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.beans.InvalidationListener;
+import javafx.geometry.Point2D;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.TextField;
+import javafx.scene.image.WritableImage;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.MouseButton;
+import javafx.scene.layout.Background;
+import javafx.scene.layout.BackgroundFill;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.StackPane;
+import javafx.scene.paint.Color;
+import javafx.scene.robot.Robot;
+import javafx.stage.Screen;
+import javafx.stage.Stage;
+import javafx.stage.StageStyle;
+import javafx.stage.WindowEvent;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.Assert;
+import junit.framework.AssertionFailedError;
+import test.util.Util;
+
+import static javafx.scene.paint.Color.MAGENTA;
+import static org.junit.Assert.fail;
+
+/**
+ * Tests to verify that the native robot implementations all work correctly.
+ */
+public class RobotTest {
+    static CountDownLatch startupLatch = new CountDownLatch(1);
+    static volatile Stage stage;
+    static volatile Scene scene;
+    static Robot robot;
+    private static final double TOLERANCE = 0.07;
```

Comment:
Can you please provide some info why this tolerance is needed?

Reply:
I gues that some tests for multi-screen configuration may be useful, because I feel that implementation of getPixel()/screenCapture are incorrect.

Reply:
TOLERANCE is needed because on macOS the pixel color is affected by the "color profile" of the display.

Reply:
I can add that as a comment, though.

Reply:
I note that all of our pixel-based tests use a similar tolerance for the same reason.

I agree that we should add some tests for multi-screen configuration, but that can be done as part of  a follow-up issue. FWIW, I tested at least one case of multiple screens with an earlier version and it worked fine.